### PR TITLE
Gives chemicals a default turf reaction which leaves a colored mess to clean up!

### PR DESCRIPTION
**Pull request by KoboldCommando**
## About The Pull Request

This adds a default reaction_turf to all reagents, which creates a cleanable turf decal that inherits the color of the reagent.  So anything you splash on a tile that doesn't have a unique reaction_turf defined will now leave a colorful mess!  Chemicals with a reaction_turf defined are unaffected.  This also applies to smoke and foam.  See included image, in the chemistry room is various individual chems, the hall to the south is the aftermath of a smoke reaction, and the room to the right is the aftermath of foam.

image

Currently it re-uses the crayon splatter sprite.  Future additions could include differentiating between liquid and solid/powder reagents, and having dynamic sprites similar to dirt.  This would require new greyscale sprites to be made.

Parent calls for colorful reagent, carpet, and holy water have been removed.  Those were the only chemicals that actually called the parent reaction_turf function, and that function was previously empty.  This ensures they won't leave any splatter behind (since their job is to make a different sort of mess, anyway).  Future chemicals will be able to decide whether or not they leave behind a mess based on whether they call the parent function.

## Why It's Good For The Game

This will give janitors a lot more to do and make their presence and activity more obvious and (hopefully) appreciated.  It will also will leave more evidence of chemical reactions and past antics and scuffles, and generally leave the station looking a hell of a lot more "lived in".  It could have practical uses for tracking down crimes, and makes the use of chemicals and grenades a bit harder to cover up, similar to how botany can be tracked down due to the splats left by their plants.

## Changelog
:cl:
add: Added a default mess that chemicals create when they're splashed on a tile.  This inherits the chemical's color, and also applies to smoke and foam!
/:cl:

### CODE REVIEW

**Comment by KoboldCommando** [comment]
One side effect of this is that quick-and-dirty cleaning smoke grenades, involving space cleaner, phosphorus, potassium and sugar, will now leave phosphorus splatters behind.  I feel like this is a minor quirk if anything, as I've not witnessed anyone other than myself ever use cleaning smoke.  Cleaning foam (as found in janitor grenades) works just fine, as does cleaning smoke with stabilized smoke powder.  If anything this could encourage a bit more use of foam and stabilized smoke powder if people are concerned about messes, which I'm personally fine with.

**Comment by KoboldCommando** [comment]
We now have custom sprites!  Only cleanable/puddle gets used, but puddlesmall and puddlesplat are included in case of any future need for recolorable liquid messes!

The messes also now have a randomized offset and rotation, and a size scale that's both randomly varied and scales off the volume of the mess!

There are a lot of ways this can be reworked or added to, but I feel like it's in a pretty good spot to provide groundwork for future patches or systems, and adding more would start to get into the realm of things like a proper liquid system.

unknown

**Comment by nednaZ** [comment]
Works as intended.

File: code/game/objects/effects/decals/cleanable/misc.dm [hunk 1]
```diff
@@ -218,3 +218,24 @@
 	icon = 'icons/effects/blood.dmi'
 	icon_state = "xfloor1"
 	random_icon_states = list("xfloor1", "xfloor2", "xfloor3", "xfloor4", "xfloor5", "xfloor6", "xfloor7")
+
+//monkestation edit begin
```
Comment (nednaZ) [inline code/game/objects/effects/decals/cleanable/misc.dm:222]: This section could be modular

Comment (KoboldCommando) [reply code/game/objects/effects/decals/cleanable/misc.dm]: Done!  decals/cleanable/misc.dm should now be untouched.  The decals are now in the monkestation folder.

**Review by nednaZ — APPROVED** [review]
Looks good to me

**Pull request merged**